lua/ide/components/bufferlist/component.lua

```diff
@@ -150,13 +173,22 @@ BufferListComponent.new = function(name, config)
                     icon = icon_set.get_icon("Folder")
                 end
             end
-            return {
-                name = libbuf.get_unique_filename(vim.api.nvim_buf_get_name(buf)),
-                icon = icon,
-                id = buf,
-            }
-        end, libbuf.get_listed_bufs())
-        self.bufs = bufs
+            if (cur_buf == buf) and self.config.current_buffer_top then
+                self.bufs[1] = {
+                    name = libbuf.get_unique_filename(vim.api.nvim_buf_get_name(buf)),
+                    icon = icon,
+                    id = buf,
+                    is_current = (cur_buf == buf)
+                }
+            else
+                table.insert(self.bufs, {
+                    name = libbuf.get_unique_filename(vim.api.nvim_buf_get_name(buf)),
+                    icon = icon,
+                    id = buf,
+                    is_current = (cur_buf == buf)
+                })
+            end
+        end
```

Comment:
isn't this logic going to end up overwriting one of the buffers? I'd recommend just adding them all to the list like we were before, then sorting by `is_current`

Reply:
Why would we overwrite? In the case where we are floating the current buffer, won't the sentinel value take up one space, until it's overwritten by the current buffer?

Reply:
because we're `table.insert`ing, but then if it's the current buf (which is not necessarily the first item in the list we're iterating over) we set `bufs[1]` which may have already been populated via `table.insert`, no? 

Unless I'm misunderstanding.

Reply:
if `self.config.current_buffer_top` is true, we are `table.inserting`ing on a table which has [1] already filled with a sentinel value, so it starts at [2] on inserting.